src/Sylius/Bundle/CoreBundle/Validator/Constraints/HasAllVariantPricesDefinedValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class HasAllVariantPricesDefinedValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($product, Constraint $constraint)
+    {
+        Assert::isInstanceOf($constraint, HasAllVariantPricesDefined::class);
+
+        if ($product->isSimple()) {
+            return;
+        }
+
+        $channels = $product->getChannels();
+
+        /** @var ProductVariantInterface $variant */
+        foreach ($product->getVariants() as $productVariant) {
+            /** @var ChannelInterface $channel */
+            foreach ($channels as $channel) {
```

Comment:
I would love to, however as we use `return` to stop validation on first invalid value, it would require to throw exception and catching it, so probably two more methods to keep one level of indentation - I don't know is it worth for such a simple case :(